keycloak 21 does not build with this spi

I started getting exceptions when building keycloak 21 with this spi:
```bash
2023-03-01 12:16:31,767 TRACE [io.quarkus.builder] (build-1) Finished step "io.quarkus.deployment.steps.MainClassBuildStep#build" in 0 ms
2023-03-01 12:16:31,767 TRACE [io.quarkus.builder] (build-6) Finished step "io.quarkus.deployment.steps.ReflectionDiagnosticProcessor#writeReflectionData" in 0 ms
2023-03-01 12:16:31,767 TRACE [io.quarkus.builder] (build-1) Dependency of "io.quarkus.deployment.pkg.steps.JarResultBuildStep#buildRunnerJar" finished; 1 remaining
2023-03-01 12:16:31,767 TRACE [io.quarkus.builder] (build-6) Dependency of "io.quarkus.deployment.pkg.steps.JarResultBuildStep#buildRunnerJar" finished; 0 remaining
2023-03-01 12:16:31,767 TRACE [io.quarkus.builder] (build-1) Starting step "io.quarkus.deployment.pkg.steps.JarResultBuildStep#buildRunnerJar"
2023-03-01 12:16:31,767 TRACE [io.quarkus.builder] (build-1) Finished step "io.quarkus.deployment.pkg.steps.JarResultBuildStep#buildRunnerJar" in 0 ms
2023-03-01 12:16:31,767 TRACE [io.quarkus.builder] (build-1) Dependency of "io.quarkus.deployment.pkg.steps.JarResultBuildStep#jarOutput" finished; 0 remaining
2023-03-01 12:16:31,767 TRACE [io.quarkus.builder] (build-1) Dependency of "io.quarkus.deployment.pkg.steps.AppCDSBuildStep#build" finished; 0 remaining
2023-03-01 12:16:31,767 TRACE [io.quarkus.builder] (build-6) Starting step "io.quarkus.deployment.pkg.steps.JarResultBuildStep#jarOutput"
2023-03-01 12:16:31,767 TRACE [io.quarkus.builder] (build-6) Finished step "io.quarkus.deployment.pkg.steps.JarResultBuildStep#jarOutput" in 0 ms
2023-03-01 12:16:31,767 TRACE [io.quarkus.builder] (build-16) Starting step "io.quarkus.deployment.pkg.steps.AppCDSBuildStep#build"
2023-03-01 12:16:31,767 TRACE [io.quarkus.builder] (build-6) End step completed; 1 remaining
2023-03-01 12:16:31,767 TRACE [io.quarkus.builder] (build-16) Finished step "io.quarkus.deployment.pkg.steps.AppCDSBuildStep#build" in 0 ms
2023-03-01 12:16:31,767 TRACE [io.quarkus.builder] (build-16) End step completed; 0 remaining
2023-03-01 12:16:31,767 TRACE [org.jboss.threads] (build-16) Thread "Thread[build-16,5,]" exiting
2023-03-01 12:16:31,767 TRACE [org.jboss.threads] (build-30) Thread "Thread[build-30,5,]" exiting
2023-03-01 12:16:31,767 TRACE [org.jboss.threads] (build-21) Thread "Thread[build-21,5,]" exiting
2023-03-01 12:16:31,767 TRACE [org.jboss.threads] (build-20) Thread "Thread[build-20,5,]" exiting
2023-03-01 12:16:31,767 TRACE [org.jboss.threads] (build-15) Thread "Thread[build-15,5,]" exiting
2023-03-01 12:16:31,767 TRACE [org.jboss.threads] (build-10) Thread "Thread[build-10,5,]" exiting
2023-03-01 12:16:31,767 TRACE [org.jboss.threads] (build-11) Thread "Thread[build-11,5,]" exiting
2023-03-01 12:16:31,767 TRACE [org.jboss.threads] (build-6) Thread "Thread[build-6,5,]" exiting
2023-03-01 12:16:31,767 TRACE [org.jboss.threads] (build-26) Thread "Thread[build-26,5,]" exiting
2023-03-01 12:16:31,767 TRACE [org.jboss.threads] (build-5) Thread "Thread[build-5,5,]" exiting
2023-03-01 12:16:31,767 TRACE [org.jboss.threads] (build-12) Thread "Thread[build-12,5,]" exiting
2023-03-01 12:16:31,767 TRACE [org.jboss.threads] (build-17) Thread "Thread[build-17,5,]" exiting
2023-03-01 12:16:31,767 TRACE [org.jboss.threads] (build-1) Thread "Thread[build-1,5,]" exiting
2023-03-01 12:16:31,767 TRACE [org.jboss.threads] (build-9) Thread "Thread[build-9,5,]" exiting
2023-03-01 12:16:31,767 TRACE [org.jboss.threads] (build-19) Thread "Thread[build-19,5,]" exiting
2023-03-01 12:16:31,767 TRACE [org.jboss.threads] (build-7) Thread "Thread[build-7,5,]" exiting
2023-03-01 12:16:31,767 TRACE [org.jboss.threads] (build-24) Thread "Thread[build-24,5,]" exiting
2023-03-01 12:16:31,767 TRACE [org.jboss.threads] (build-22) Thread "Thread[build-22,5,]" exiting
2023-03-01 12:16:31,767 TRACE [org.jboss.threads] (build-2) Thread "Thread[build-2,5,]" exiting
2023-03-01 12:16:31,768 TRACE [org.jboss.threads] (build-23) Thread "Thread[build-23,5,]" exiting
2023-03-01 12:16:31,768 TRACE [org.jboss.threads] (build-3) Thread "Thread[build-3,5,]" exiting
2023-03-01 12:16:31,768 TRACE [org.jboss.threads] (build-25) Thread "Thread[build-25,5,]" exiting
2023-03-01 12:16:31,768 TRACE [org.jboss.threads] (build-29) Thread "Thread[build-29,5,]" exiting
2023-03-01 12:16:31,768 TRACE [org.jboss.threads] (build-4) Thread "Thread[build-4,5,]" exiting
2023-03-01 12:16:31,768 TRACE [org.jboss.threads] (build-28) Thread "Thread[build-28,5,]" exiting
2023-03-01 12:16:31,768 TRACE [org.jboss.threads] (build-27) Thread "Thread[build-27,5,]" exiting
2023-03-01 12:16:31,768 TRACE [org.jboss.threads] (build-18) Thread "Thread[build-18,5,]" exiting
2023-03-01 12:16:31,768 TRACE [org.jboss.threads] (build-8) Thread "Thread[build-8,5,]" exiting
2023-03-01 12:16:31,768 TRACE [org.jboss.threads] (build-14) Thread "Thread[build-14,5,]" exiting
2023-03-01 12:16:31,768 TRACE [org.jboss.threads] (build-13) Thread "Thread[build-13,5,]" exiting
2023-03-01 12:16:31,768 TRACE [org.jboss.threads] (build-31) Thread "Thread[build-31,5,]" exiting
ERROR: Failed to run 'build' command.
Error details:
java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doReaugment(QuarkusEntryPoint.java:84)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:48)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:32)
	at org.keycloak.quarkus.runtime.cli.command.Build.run(Build.java:71)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1939)
	at picocli.CommandLine.access$1300(CommandLine.java:145)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2358)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2352)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2314)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2316)
	at picocli.CommandLine.execute(CommandLine.java:2078)
	at org.keycloak.quarkus.runtime.cli.Picocli.runReAugmentation(Picocli.java:183)
	at org.keycloak.quarkus.runtime.cli.Picocli.runReAugmentationIfNeeded(Picocli.java:120)
	at org.keycloak.quarkus.runtime.cli.Picocli.parseAndRun(Picocli.java:88)
	at org.keycloak.quarkus.runtime.KeycloakMain.main(KeycloakMain.java:88)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:61)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:32)
Caused by: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step org.keycloak.quarkus.deployment.KeycloakProcessor#setCryptoProvider threw an exception: java.lang.LinkageError: loader constraint violation: loader io.quarkus.bootstrap.classloading.QuarkusClassLoader @14faa38c wants to load class org.keycloak.common.crypto.FipsMode. A different class with the same name was previously loaded by java.net.URLClassLoader @614df0a4. (org.keycloak.common.crypto.FipsMode is in unnamed module of loader java.net.URLClassLoader @614df0a4, parent loader 'app')
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1012)
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:495)
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:455)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:467)
	at io.quarkus.deployment.recording.BytecodeRecorderImpl$$RecordingProxyProxy6.setCryptoProvider(Unknown Source)
	at org.keycloak.quarkus.deployment.KeycloakProcessor.setCryptoProvider(KeycloakProcessor.java:636)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:281)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
	at java.base/java.lang.Thread.run(Thread.java:833)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:335)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createProductionApplication(AugmentActionImpl.java:175)
	at io.quarkus.deployment.mutability.ReaugmentTask.main(ReaugmentTask.java:69)
	... 26 more
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step org.keycloak.quarkus.deployment.KeycloakProcessor#setCryptoProvider threw an exception: java.lang.LinkageError: loader constraint violation: loader io.quarkus.bootstrap.classloading.QuarkusClassLoader @14faa38c wants to load class org.keycloak.common.crypto.FipsMode. A different class with the same name was previously loaded by java.net.URLClassLoader @614df0a4. (org.keycloak.common.crypto.FipsMode is in unnamed module of loader java.net.URLClassLoader @614df0a4, parent loader 'app')
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1012)
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:495)
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:455)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:467)
	at io.quarkus.deployment.recording.BytecodeRecorderImpl$$RecordingProxyProxy6.setCryptoProvider(Unknown Source)
	at org.keycloak.quarkus.deployment.KeycloakProcessor.setCryptoProvider(KeycloakProcessor.java:636)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:281)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
	at java.base/java.lang.Thread.run(Thread.java:833)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
	at io.quarkus.builder.Execution.run(Execution.java:123)
	at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:79)
	at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:160)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:331)
	... 28 more
Caused by: java.lang.LinkageError: loader constraint violation: loader io.quarkus.bootstrap.classloading.QuarkusClassLoader @14faa38c wants to load class org.keycloak.common.crypto.FipsMode. A different class with the same name was previously loaded by java.net.URLClassLoader @614df0a4. (org.keycloak.common.crypto.FipsMode is in unnamed module of loader java.net.URLClassLoader @614df0a4, parent loader 'app')
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1012)
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:495)
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:455)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:467)
	at io.quarkus.deployment.recording.BytecodeRecorderImpl$$RecordingProxyProxy6.setCryptoProvider(Unknown Source)
	at org.keycloak.quarkus.deployment.KeycloakProcessor.setCryptoProvider(KeycloakProcessor.java:636)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:281)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
	at java.base/java.lang.Thread.run(Thread.java:833)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
The command '/bin/sh -c /opt/keycloak/bin/kc.sh -v build' returned a non-zero code: 1
Cleaning up project directory and file based variables
00:01
ERROR: Job failed: exit code 1
```

It's hard to establish whether this error is actually caused by a peculiarity of the spi implementation or by some bug in the keycloak itself. The error with fips module is confusing to me - we don't use it and never activated it.

I created an [issue](https://github.com/keycloak/keycloak/issues/17370) in the keycloak repository about this and one of the contributors suggested that the problem is in the spi code itself. 
Could you comment on this? If the problem is really with the spi implementation, could you fix it?

P.S.: https://github.com/keycloak/keycloak/issues/16202 is similar to this problem.

--- a/README.md
+++ b/README.md
@@ -17,9 +17,9 @@ Simple module for [Keycloak](https://www.keycloak.org/) to produce keycloak even
 
 Kafka version: `2.12-2.1.x`, `2.12-2.4.x`, `2.12-2.5.x`, `2.13-2.8`, `2.13-3.3.x`
 
-Keycloak version: `19.0.x`
+Keycloak version: `19.0.x, 21.0.x`
 
-Java version: `11`, `17`
+Java version: `17`
 
 Check out [this older version](https://github.com/SnuK87/keycloak-kafka/tree/1.1.1) to run the module on a Wildfly server
 

--- a/README.md
+++ b/README.md
@@ -47,7 +47,7 @@ The following properties can be set via environment variables (e.g. `${KAFKA_TOP
 
 - `clientId` (env `KAFKA_CLIENT_ID`): The `client.id` used to identify the client in kafka.
 
-- `bootstrapServer` (env `KAFKA_BOOTSTRAP_SERVERS`): A comma separated list of available brokers.
+- `bootstrapServers` (env `KAFKA_BOOTSTRAP_SERVERS`): A comma separated list of available brokers.
 
 - `events` (env `KAFKA_EVENTS`): The events that will be send to kafka.
 

--- a/pom.xml
+++ b/pom.xml
@@ -4,12 +4,12 @@
 	<modelVersion>4.0.0</modelVersion>
 	<groupId>com.github.snuk87.keycloak</groupId>
 	<artifactId>keycloak-kafka</artifactId>
-	<version>1.1.2</version>
+	<version>1.1.3</version>
 
 	<properties>
 		<maven.compiler.source>17</maven.compiler.source>
 		<maven.compiler.target>17</maven.compiler.target>
-		<keycloak.version>20.0.3</keycloak.version>
+		<keycloak.version>21.0.1</keycloak.version>
 		<kafka.version>3.3.1</kafka.version>
 		<junit.version>5.8.1</junit.version>
 	</properties>
@@ -19,6 +19,7 @@
 			<groupId>org.keycloak</groupId>
 			<artifactId>keycloak-core</artifactId>
 			<version>${keycloak.version}</version>
+			<scope>provided</scope>
 		</dependency>
 
 		<dependency>

